futures/examples/functional.rs

```diff
@@ -0,0 +1,57 @@
+#![feature(futures_api, async_await, await_macro)]
+#[macro_use] extern crate futures;
+
+use futures::{
+    channel::mpsc,
+    executor, //standard executors to provide a context for futures and streams
+    futures::ready,
+    StreamExt
+};
+
+fn main() {
+    let (tx, rx) = mpsc::unbounded::<i32>();
+
+    //
+    // Create a future by building an async block that will be implemented as a future,
+    // whose Future::poll will be called sometime later with a std::task::Context
+    //
+    let fut_values = async {
+        //
+        // Create another async block, again that implements future and will
+        // be provided a std::task::Context at some point
+        //
+        let fut_tx_result = async {
+            (0..100).for_each(|v| {
+                tx.unbounded_send(v).expect("Failed to send");
+            })
```

Comment:
There's no reason for putting this into an async block

Reply:
It is being spawned off to a separate task though.

Reply:
Hmm yes, you're right